scientific_library/tvb/simulator/plot/power_spectra_interactive.py

```diff
@@ -156,7 +158,11 @@ def configure(self):
         LOG.debug("time_series shape: %s" % str(self.time_series.data.shape))
         #TODO: if isinstance(self.time_series, TimeSeriesSurface) and self.first_n == -1: #LOG.error, return.
         self.data = self.time_series.data[:, :, :self.first_n, :]
-        self.period = self.time_series.sample_period
+        if self.time_series.sample_period_unit == "s":
+            self.period = self.time_series.sample_period
+        else 
+            self.period = 1 / self.time_series.sample_rate  #JC: convert sampling period of time_series.data in second
```

Comment:
I think the value of the `sample_rate` property is always converted to seconds inside the `TimeSeries` class, so we could avoid the if-else and keep only line 164. But please correct me if I am wrong.

Reply:
yes, you are right!
I didn't see the last line code for the sample rate in seconds inside the `TimeSeries`